includes/data-events/class-data-events.php

```diff
@@ -23,6 +23,21 @@ final class Data_Events {
 	 */
 	const LOGGER_HEADER = 'NEWSPACK-DATA-EVENTS';
 
+	/**
+	 * Option name for storing the nonce.
+	 */
+	const NONCE_OPTION = 'newspack_data_events_nonce';
+
+	/**
+	 * Option name for storing the nonce expiration.
+	 */
+	const NONCE_EXPIRATION_OPTION = 'newspack_data_events_nonce_expiration';
+
+	/**
+	 * Nonce lifetime in seconds (24 hours).
+	 */
+	const NONCE_LIFETIME = 86400; // 24 hours in seconds
```

Comment:
This nonce strategy is less secure than WP's, wdyt we flush it more often? 1 hour?

Reply:
We could also do something other than nonces. Make it more secure by signing the payload, maybe an hmac hash? It's the server talking to itself, so we don't need a key pair, just the signature hash.

Reply:
Nevermind, exactly because the server is talking to itself it doesn't matter much. I'd still lower the lifetime though